controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -1089,17 +1097,23 @@ private void closeConnection(Reply reply, RawClient client) {
      * @return true if reply is in the expected reply set for the given requestType or throw exception.
      */
     @SneakyThrows(ConnectionFailedException.class)
-    private Boolean transformReply(Reply reply, RawClient client, String qualifiedStreamSegmentName, Class<? extends Request> requestType) {
+    private Boolean transformReply(Reply reply,
+                                   RawClient client,
+                                   String qualifiedStreamSegmentName,
+                                   Class<? extends Request> requestType,
+                                   WireCommandType type) {
         closeConnection(reply, client);
         Set<Class<? extends Reply>> expectedReplies = this.EXPECTED_REPLIES.get(requestType);
         if (expectedReplies != null && expectedReplies.contains(reply.getClass())) {
             log.info(reply.getRequestId(), "{} {} {}.", requestType.getSimpleName(), qualifiedStreamSegmentName,
                     reply.getClass().getSimpleName());
             return true;
         } else if (reply instanceof WireCommands.NoSuchSegment) {
-            throw new NoSuchSegmentException(reply.toString());
+            throw new WireCommandFailedException(type, WireCommandFailedException.Reason.SegmentDoesNotExist);
         } else if (reply instanceof WireCommands.WrongHost) {
-            throw new ConnectionFailedException(reply.toString());
+            throw new WireCommandFailedException(type, WireCommandFailedException.Reason.UnknownHost);
+        } else if (reply instanceof WireCommands.TableSegmentNotEmpty) {
```

Comment:
Why can't you put this in the Map you have at the top of this class?

Reply:
the map is for expectedReplies